services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java

```diff
@@ -61,15 +61,22 @@ public static final String getCurrentUser() {
     return null;
   }
 
-  public static final boolean canEditChallenge(List<Long> managersId) {
+  public static final boolean canEditChallenge(List<Long> managersId, String spaceId) {
     Identity identity = getIdentityByTypeAndId(OrganizationIdentityProvider.NAME, getCurrentUser());
+    Space space = getSpaceById(spaceId);
+    SpaceService spaceService = CommonsUtils.getService(SpaceService.class);
+    Boolean isChallengeOwner = false;
+    Boolean isSpaceManager = false;
     if (identity != null) {
-      return managersId.stream().anyMatch(i -> i == Long.parseLong(identity.getId()));
-    } else {
-      return true;
+      isChallengeOwner = managersId.stream().anyMatch(i -> i == Long.parseLong(identity.getId()));
     }
+    if (space != null) {
+      isSpaceManager = spaceService.isManager(space, getCurrentUser());
```

Comment:
We need to check if the user is a space manager or a member of the all spaces administrators
```suggestion
      isSpaceManager = spaceService.isManager(space, userId) || spaceService.isSuperManager(userId);
```